src/ClientTrait.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud;
+
+use Google\Cloud\Storage\Connection\ConnectionInterface;
+
+/**
+ * Provides shared functionality between Clients.
+ */
+trait ClientTrait
```

Comment:
Yes, I would see this trait housing more functionality in the future. Possibly logging or caching related, for example. It could definitely end up evolving into something else if that is more fitting - whatever makes the most sense as the project grows.

Would you suggest a different structure?